build-tools/packages/build-cli/src/library/repoPolicyCheck/common.ts

```diff
@@ -14,7 +14,12 @@ export interface Handler {
 	name: string;
 	match: RegExp;
 	handler: (file: string, root: string) => Promise<string | undefined>;
-	resolver?: (file: string, root: string) => { resolved: boolean; message?: string };
+	resolver?: (
+		file: string,
+		root: string,
+	) =>
+		| Promise<{ resolved: boolean; message?: string }>
+		| { resolved: boolean; message?: string };
```

Comment:
Any reason the resolvers shouldn't always be async?

Reply:
They can be, but there isn't a reason to force them to be async. If a resolver isn't returning a Promise, then requiring Promise causes build breaks. TFoo does not satisfy Promise<TFoo>. If we do want them all async, then that should be a follow-up.